api/v1alpha1/clusterextension_types.go

```diff
@@ -65,6 +65,11 @@ type ClusterExtensionSpec struct {
 	//
 	// Defines the policy for how to handle upgrade constraints
 	UpgradeConstraintPolicy UpgradeConstraintPolicy `json:"upgradeConstraintPolicy,omitempty"`
+
+	//+kubebuilder:Optional
+	//
+	// watchNamespaces indicates which namespaces the operator should watch.
```

Comment:
Nit: General GoDoc best practice is for the first word of GoDoc to be the exact name of the identifier (so in this case `WatchNamespaces`). Spot check of `kubectl explain pod.spec` seems to mostly-ish follow that pattern, although it is admittedly hit or miss.

```suggestion
	// WatchNamespaces indicates which namespaces the operator should watch.
```

Should we say something here about this being a convention that the extension needs to support?

This also reminds me. Does rukpak have any handling of this field for plain bundles? Or is this right now only something that works with registry+v1 bundles?

Reply:
While it's true that godoc docs should typically start with a capital letter, because these comments are used to generate openapi schema descriptions, the k8s convention is to start with a lowercase letter so it matches the field name in the schema.

Reply:
> This also reminds me. Does rukpak have any handling of this field for plain bundles? Or is this right now only something that works with registry+v1 bundles?

This works only for registryV1 for now, wherein we generate the RBAC based on CSV. For the other bundle formats, we just apply whatever is present in the bundle. We would have to discuss further on how evaluate the operators' RBAC and validate if specified targetNs is in sync with what the operator supports. (have created an issue in rukpak: https://github.com/operator-framework/rukpak/issues/815). Probably use SAR? We can follow up in the other issue. 

Given this is only for registryV1 for now I don't think we are there yet to say that all extensions need to support this feature imo. Thoughts?

Reply:
Yeah, we need to think about how this would work for other bundle types. For now, would it be reasonable for the field description to explicitly call out that this configuration only applies to registry+v1 bundles?

Reply:
Done.